librad/src/git/storage.rs

```diff
@@ -193,7 +236,14 @@ impl Storage {
 
         match reference {
             None => Ok(None),
-            Some(r) => Ok(Some(r.peel_to_commit()?.id().into())),
+            Some(r) => r
+                .peel(kind)
+                .map(Some)
+                .or_matches::<Error, _, _>(is_invalid_spec_err, || {
```

Comment:
Sorry for the persistent nitpicking, but I think this is perhaps not the best idea. We use `Result<Option<T>, E>` where `Ok(None)` means "doesn't exist". A type mismatch is clearly something I'd expect to be an error. Let's just say `tip` gives me the `Object` the ref points to, or `None` if the reference doesn't exist.

Reply:
Ya fair, my thinking was that `None` meant that the `Object` doesn't exist for the type supplied. But you're right, the user messed up so the error should be the loud and proud.